test/integration/integration_test.cc

```diff
@@ -1977,4 +1977,74 @@ TEST_P(IntegrationTest, RandomPreconnect) {
   }
 }
 
+// Tests that a filter (set-route-filter) using the setRoute callback and DelegatingRoute mechanism
+// successfully overrides the cached route, and subsequently, the request's upstream cluster
+// selection.
+TEST_P(IntegrationTest, SetRouteToDelegatingRouteWithClusterOverride) {
+  useAccessLog("%UPSTREAM_CLUSTER%\n");
+
+  config_helper_.addFilter(R"EOF(
+    name: set-route-filter
+    )EOF");
+
+  setUpstreamCount(2);
+
+  // Tests with ORIGINAL_DST cluster because the first use case of the setRoute / DelegatingRoute
+  // route mutability functionality will be for a filter that re-routes requests to an
+  // ORIGINAL_DST cluster on a per-request basis.
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    std::string cluster_yaml = R"EOF(
+            name: cluster_override
+            connect_timeout: 1.250s
+            type: ORIGINAL_DST
+            lb_policy: CLUSTER_PROVIDED
+            original_dst_lb_config:
+              use_http_header: true
+          )EOF";
+    envoy::config::cluster::v3::Cluster cluster_config;
+    TestUtility::loadFromYaml(cluster_yaml, cluster_config);
+    auto* orig_dst_cluster = bootstrap.mutable_static_resources()->add_clusters();
+    orig_dst_cluster->MergeFrom(cluster_config);
+  });
+
+  auto co_vhost =
+      config_helper_.createVirtualHost("cluster_override vhost", "/some/path", "cluster_override");
+  config_helper_.addVirtualHost(co_vhost);
+
+  initialize();
+
+  const std::string ip_port_pair =
+      absl::StrCat(Network::Test::getLoopbackAddressString(GetParam()), ":",
```

Comment:
if you use `Network::Test::getLoopbackAddressUrlString` instead of `Network::Test::getLoopbackAddressString` `getLoopbackAddressUrlString` it should pass on windows

Reply:
thanks for the pointer! you are right, I'm missing [the brackets needed](https://github.com/envoyproxy/envoy/blob/main/test/test_common/network_utility.cc#L74-L79) for IPv6 

trying out now (92f250d)